### ISSUE TITLE
Various library bundling nits

### ISSUE THREAD

**Issue by KristofferC** [JuliaLang/julia#46872]
####  Multiple copies of the same library on Windows:

On Windows we seem to concretize symlinks into copies of the libraries which leads to duplication:

```
-rw------- 1 kc kc 2.2M Sep  9 11:36 libblastrampoline-5-0-2.dll
-rw------- 1 kc kc 2.2M Sep  9 11:36 libblastrampoline-5.dll
-rw------- 1 kc kc 2.2M Sep  9 11:36 libblastrampoline.dll
```

We only use `libblastrampoline.dll` anyway.

#### Different `libpcre2` versions shipped on Windows:

```
-rw------- 1 kc kc 689K Sep  9 11:37 libpcre2-16-0.dll
-rw------- 1 kc kc 689K Sep  9 11:37 libpcre2-16.dll
-rw------- 1 kc kc 661K Sep  9 11:37 libpcre2-32-0.dll
-rw------- 1 kc kc 661K Sep  9 11:37 libpcre2-32.dll
-rw------- 1 kc kc 751K Sep  9 11:37 libpcre2-8-0.dll
-rw------- 1 kc kc 751K Sep  9 11:37 libpcre2-8.dll
-rw------- 1 kc kc 112K Sep  9 11:37 libpcre2-posix-3.dll
```

but I think we only use `libpcre2-8.dll`.

####  Some LLVM support libs only shipped on Windows:

```
-rw------- 1 kc kc 293K Sep  9 11:37 libRemarks.dll
-rw------- 1 kc kc 390K Sep  9 11:37 libLTO.dll
-rw------- 1 kc kc 326K Sep  9 11:37 libmlir_async_runtime.dll
-rw------- 1 kc kc 901K Sep  9 11:37 libmlir_c_runner_utils.dll
-rw------- 1 kc kc 323K Sep  9 11:37 libmlir_runner_utils.dll
``` 

are they needed?

####  Inconsisteny in naming of `libLLVM`:

```
libLLVM.dylib
libLLVM-13jl.so 
libLLVM-13jl.dll
```

Maybe mac should have `-13jl` as well?

**Comment by giordano** [comment]
> We only use `libblastrampoline.dll` anyway. 

That's wrong though: https://github.com/JuliaLinearAlgebra/libblastrampoline/issues/87

**Comment by inkydragon** [comment]
> On Windows we seem to concretize symlinks into copies of the libraries which leads to duplication:

By default, `ln -s` copies the file on windows instead of creating a symbolic link.
Administrator privileges and an environment variable are required to create a real symbolic link.
A possible alternative is to create a hard link. `cp -f --dereference --link src dst`

- https://github.com/JuliaLinearAlgebra/libblastrampoline/blob/dbb8221eb0c9ecc7eea0ad57677b68965299032c/src/Makefile#L45-L52
- cygwin: https://stackoverflow.com/questions/3648819/how-to-make-a-symbolic-link-with-cygwin-in-windows-7
- MSYS2: https://stackoverflow.com/questions/61594025/symlink-in-msys2-copy-or-hard-link

**Comment by KristofferC** [comment]
> A possible alternative is to create a hard link. cp -f --dereference --link src dst

Is there a need for any links at all? `libblastrampoline` seems to be the odd one out here of all the libraries on Windows with three copies.

> Those DLLs came from: https://github.com/JuliaBinaryWrappers/LLVM_jll.jl

Are they needed for anything in the Julia installation?

**Comment by giordano** [comment]
With #47676 there will be a single copy of libblastrampoline on Windows, the different copies were error-prone anyway.
